File: java/src/juegoTexto/VistaTextual.java
package juegoTexto;

import civitas.CivitasJuego;
import civitas.Diario;
import civitas.OperacionesJuego;
import civitas.Respuestas;
import civitas.SalidasCarcel;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Scanner;
import civitas.Casilla;
import civitas.Jugador;
import civitas.TituloPropiedad;

class VistaTextual {
  
  CivitasJuego juegoModel; 
  int iGestion=-1;
  int iPropiedad=-1;
  private static String separador = "=====================";
  
  private Scanner in;
  
  VistaTextual () {
    in = new Scanner (System.in);
  }
  
  void mostrarEstado(String estado) {
    System.out.println (estado);
  }
              
  void pausa() {
    System.out.print ("Pulsa una tecla");
    in.nextLine();
  }

  int leeEntero (int max, String msg1, String msg2) {
    Boolean ok;
    String cadena;
    int numero = -1;
    do {
      System.out.print (msg1);
      cadena = in.nextLine();
      try {  
        numero = Integer.parseInt(cadena);
        ok = true;
      } catch (NumberFormatException e) { // No se ha introducido un entero
        System.out.println (msg2);
        ok = false;  
      }
      if (ok && (numero < 0 || numero >= max)) {
        System.out.println (msg2);
        ok = false;
      }
    } while (!ok);

    return numero;
  }

  int menu (String titulo, ArrayList<String> lista) {
    String tab = "  ";
    int opcion;
    System.out.println (titulo);
    for (int i = 0; i < lista.size(); i++) {
      System.out.println (tab+i+"-"+lista.get(i));
    }

    opcion = leeEntero(lista.size(),
                       "\n"+tab+"Elige una opción: ",
                       tab+"Valor erróneo");
    return opcion;
  }

  SalidasCarcel salirCarcel() {
    int opcion = menu ("Elige la forma para intentar salir de la carcel",
      new ArrayList<> (Arrays.asList("Pagando","Tirando el dado")));
    return (SalidasCarcel.values()[opcion]);
  }

  Respuestas comprar() {
    int opcion = menu("Desea comprar la calle?", 
      new ArrayList<>(Arrays.asList("No","Si")));
    return (Respuestas.values()[opcion]);
  }

  void gestionar () {
    int opcion = menu("Desea realizar alguna gestion?",
      new ArrayList<>(Arrays.asList("Vender propiedad","Hipotecar propiedad",
      "Cancelar una hipoteca","Construir casa","Construir hotel","Terminar")));
    iGestion=opcion;
    //Propiedad jugador actual?
    
  }
  
  public int getGestion(){return iGestion;}
  
  public int getPropiedad(){return iPropiedad;}
    

  void mostrarSiguienteOperacion(OperacionesJuego operacion) {}


  void mostrarEventos() {
    while(Diario.getInstance().eventosPendientes())
      mostrarEstado(Diario.getInstance().leerEvento());
  }
  
  public void setCivitasJuego(CivitasJuego civitas){ 
        juegoModel=civitas;
        this.actualizarVista();
    }
  
  void actualizarVista(){
  
  } 
}

File: ruby/lib/vista_textual.rb
#encoding:utf-8
require_relative 'operaciones_juego'
require_relative 'salidas_carcel'
require 'io/console'

module Civitas

  class Vista_textual
    #Actualizar vista 
    #Controlador::Juega
    def initialize
      @in
      @iPropiedad = 0
      @iGestion = 0
      @juegoModel
    end

    def mostrar_estado(estado)
      puts estado
    end

    
    def pausa
      print "Pulsa una tecla"
      gets
      print "\n"
    end

    def lee_entero(max,msg1,msg2)
      ok = false
      begin
        print msg1
        cadena = gets.chomp
        begin
          if (cadena =~ /\A\d+\Z/)
            numero = cadena.to_i
            ok = true
          else
            raise IOError
          end
        rescue IOError
          puts msg2
        end
        if (ok)
          if (numero >= max)
            ok = false
          end
        end
      end while (!ok)

      return numero
    end



    def menu(titulo,lista)
      tab = "  "
      puts titulo
      index = 0
      lista.each { |l|
        puts tab+index.to_s+"-"+l
        index += 1
      }

      opcion = lee_entero(lista.length,
                          "\n"+tab+"Elige una opción: ",
                          tab+"Valor erróneo")
      return opcion
    end

    def salirCarcel
      lista_Respuestas = [SalidasCarcel::PAGANDO, SalidasCarcel::TIRANDO]
      return lista_Respuestas[menu("¿Cómo quieres salir de la cárcel?", ["PAGANDO", "TIRANDO"])]
    end
    
    def comprar
      lista_Respuestas = [Respuestas::NO, Respuestas::Si]
      return lista_Respuestas[menu("¿Quieres comprar la calle?", ["NO", "SI"])]
    end

    def gestionar
      @iGestion = menu("Número de gestión de inmobiliaria", ["VENDER", "HIPOTECAR", "CANCELAR_HIPOTECA", "CONSTRUIR_CASA", "CONSTRUIR_CASA", "CONSTRUIR_HOTEL", "TERMINAR"])
      @iPropiedad = menu("Número de la propiedad", @juego.getJugadorActual.propiedades)
    end

    def getGestion
      return @iGestion
    end

    def getPropiedad
      return @iPropiedad
    end

    def mostrarSiguienteOperacion(operacion)
      case operacion
      when Operaciones_juego::PASAR_TURNO
        puts "PASAR_TURNO"
      when Operaciones_juego::SALIR_CARCEL
        puts "SALIR_CARCEL"
      when Operaciones_juego::AVANZAR
        puts "AVANZAR"
      when Operaciones_juego::COMPRAR
        puts "COMPRAR"
      when Operaciones_juego::GESTIONAR
        puts "GESTIONAR"
      end

    end

    def mostrarEventos
      while Diario.instance.eventosPendientes
        puts Diario.instance.leerEvento
      end
    end

    def setCivitasJuego(civitas)
         @juegoModel=civitas
         self.actualizarVista
    end

    def actualizarVista
      @juegoModel.infoJugadorTexto
    end

    
  end

end
